src/main/java/core/basesyntax/MatrixOperation.java

```diff
@@ -5,15 +5,46 @@
  * максимального та мінімального елементів.
  */
 public class MatrixOperation {
-    public int[] getDiagonal(int[][] matrix) {
-        return null;
+    protected int[] getDiagonal(int[][] matrix) {
+        int row = matrix.length;
+        int col = matrix[0].length;
+        int[] result;
+        if (row < col) {
+            result = new int[row];
+        } else {
+            result = new int[col];
+        }
+        for (int i = 0; i < row; i++) {
+            for (int j = 0; j < col; j++) {
+                if (i == j) {
+                    result[i] = matrix[i][j];
+                }
+            }
+        }
+        return result;
     }
 
-    public int getMin(int[][] matrix) {
-        return 0;
+    protected int getMin(int[][] matrix) {
+        int minimum = Integer.MAX_VALUE;
```

Comment:
why we use Integer.MAX_VALUE?

Reply:
@SviatoslavExpert lets use first element of matrix as we have discussed on the lesson

Reply:
Fixed.  Thank you.